### PR TITLE
k8s migration

### PR DESCRIPTION
**Pull request by xMHW**
# Summary 
Migration of zabo backend to sparcs-cluster.
Major changes
- environment variables managed with zabo-server-deployment/base/server/configmap.yaml and sealed-secret.yaml
- deploy to EKS based Sparcs-Cluster
- gitops supported based on argocd, definitions under zabo-server-deployment folder

# Further Work 
- [ ] Migrate mongoDB data to newly deployed mongoDB.
- [ ] zabo-server-deployment/base/server/secret-template.yaml using this template, change the according base64value and use kubeseal to update sealed-secret.yaml (https://www.notion.so/sparcs/K8S-Sealed-Secret-kubeseal-c3e315e429c442bebf8998b048404e17?pvs=4).
- [ ] zabo-server-deployment/base/server/configmap.yaml change the value of configmap variables accordingly.
- [ ] change the domain of ingress zabo-server-deployment/base/server/ingress.yaml after updating domain.
- [ ] build docker image and upload to ECR(rather than github registry) and change zabo-server-deployment/base/server/deployment.yaml image configuration and zabo-server-deployment/overlays/dev/kustomization.yaml this too.
- [ ] Need to setup mongoDB backup job(This could be done by wheel members after discussion).

Please Squash Merge this PR please!
🚀 LGTM :D

### CODE REVIEW

File: zabo-server-deployment/base/server/sealed-secret.yaml
Comment (xMHW) [inline zabo-server-deployment/base/server/sealed-secret.yaml:1]: 환경변수 업데이트 필요 @SnowSuno @withSang

File: zabo-server-deployment/base/server/configmap.yaml
Comment (xMHW) [inline zabo-server-deployment/base/server/configmap.yaml:1]: 환경변수 업데이트 필요 @SnowSuno @withSang

File: zabo-server-deployment/base/server/deployment.yaml
Comment (xMHW) [inline zabo-server-deployment/base/server/deployment.yaml:1]: ghcr.io -> ECR migration 필요 @SnowSuno @withSang

File: zabo-server-deployment/base/server/sealed-secret.yaml
Comment (xMHW) [reply zabo-server-deployment/base/server/sealed-secret.yaml:1]: [kubeseal](https://www.notion.so/sparcs/K8S-Sealed-Secret-kubeseal-c3e315e429c442bebf8998b048404e17) 문서 참조

**Review by withSang — APPROVED** [review]
우왕 LGTM임돠!! 고생 많으셨습니다! 👍 
이미지 레지스트리 변경, configmap 변경, sealedsecret 설정 모두 머지한 뒤에 진행할게요!

**Pull request merged**